sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -52,21 +53,39 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     }
   }
 
+  private def cache(
+      relation: DataSourceV2Relation,
+      tableName: String,
+      storageLevel: StorageLevel): Unit = {
+    session.sharedState.cacheManager.cacheQuery(
+      session,
+      relation,
+      Some(tableName),
+      storageLevel)
```

Comment:
Maybe just call `session.sharedState.cacheManager.cacheQuery` in `RenameTableExec`. This is just one line call, seems not necessary to wrap with a method.

Reply:
The intention was to wrap `SparkSession` and take `DataSourceV2Relation` explicitly instead of `LogicalPlan`. I don't have a strong opinion here and followed your suggestion here.